neo4j-ogm-docs/src/main/asciidoc/reference/nativetypes.adoc

```diff
@@ -0,0 +1,183 @@
+Therefore it is an Opt-in feature.
+
+To opt-in, please first add the corresponding module for your <<reference:native-property-types:supported-drivers, driver>> and
+than use the new configuration property `use-native-types`:
+
+.Enabling the usage of native types
+[%autowidth.spread,frame="topbot",options="header"]
+|======================
+|ogm.properties   | Java Configuration
+
+a|
+[source, properties]
+----
+URI=bolt://neo4j:password@localhost
+use-native-types=true
+----
+
+a|
+[source, java]
+----
+Configuration configuration = new Configuration.Builder()
+        .uri("bolt://neo4j:password@localhost")
+        .useNativeTypes()
+        .build()
+----
+|======================
+
+The same works with a file based URI for the embedded driver.
+
+Once enabled, native types are used for all attributes of all node- and relationship-entities and also for all parameters passed through the OGM `Session` interface.
+
+[[reference:native-property-types:mapping]]
+== Mapping of native types
+
+The following table describes how Neo4j temporal and spatial property types are mapped to attributes of Neo4j-OGM entities:
+
+======
+[options="header", cols="m,"]
+|===
+| Neo4j type        | Neo4j-OGM type
+| Date | `LocalDate`
+| Time | `OffsetTime`
+| LocalTime | `LocalTime`
+| DateTime | `ZonedDateTime`
+| LocalDateTime | `LocalDateTime`
+| Duration | `TemporalAmount`*
+| Point | One of four different Neo4j-OGM  in `org.neo4j.ogm.types.spatial`**
+|===
+
+
++*+ The Neo4j `Duration` can either be a Java 8 `Duration` or `Period` with the least common denominator being a `TemporalAmount`.
+A Java 8 duration always deals with an exact number of seconds while periods take daylight saving times and others into account when added to instants.
+If you are sure that you only deal with one or the other, you just use an explicit mapping to either `java.time.Duration` or `java.time.Period`.
+
++**+ There is no generic Java type representing a spatial point.
+As OGM supports different ways of connecting to Neo4j it cannot expose either the Java drivers or the internal representation of a point, so it provides a point of it's own.
+Please read the next section to learn which concrete classes Neo4j-OGM offers for a point.
+======
+
+[[reference:native-property-types:spatial-types]]
+== Mapping of Neo4j spatial types
+
+Neo4j supports four slightly different property types for spatial points, see https://neo4j.com/docs/cypher-manual/current/syntax/spatial/[Spatial values].
+All variations of the `Point` type are backed by an index and therefore perform very well in queries.
+The main difference between them is the coordinate reference system.
+A point can either be stored in a geographic coordinate system with longitude and latitude or in a cartesian system with x and y.
+If you add the third dimension, you add altitude or a z-axis.
```

Comment:
In geographic CRS we add `height`, not `altitude`. Altitude implies height about sea level. See the original documents I wrote.

Here is an extract:

> In 3D we support z (height) with units ‘meters’. The WGS-84-3D CRS references the defintion at http://spatialreference.org/ref/epsg/4979/html/, which defines the z axis as being ‘ellipsoidal height’, which means height above the WGS84 ellipsoid, not gravity based height (height above the geoid modelled sea level, or elevation). The difference between these two heights can be large (for example, some systems model the difference between the two as approximately 47m in Germany). However, for most purposes in Neo4j, storing elevation in this field will not incur any negative consequences. The approximations in the 3D distance calculation are probably of more consequence than the use of elevations in the ellipsoidal height field.

Also, all points will store coordinates as `x`, `y` (and `z`). The terms `longitude`, `latitude` and `height` are only there as aliases that have the side effect of setting a different default `crs` for cases when the `crs` is not explicitly set.

Reply:
Height was a good catch! That will result in a change to the class. Thank you.